src/core/main.c

```diff
@@ -3130,6 +3140,13 @@ int main(int argc, char *argv[]) {
                 /* clear the kernel timestamp, because we are not PID 1 */
                 kernel_timestamp = DUAL_TIMESTAMP_NULL;
 
+                /* Preserve the ambient set for later use with *some* forked processes */
+                r = capability_get_ambient(&starting_ambient_set);
+                if (r < 0) {
+                        error_message = "Failed to save ambient capabilities";
+                        goto finish;
+                }
```

Comment:
seems safe to always acquire, no? i.e. no need to conditionalize this. This isn't precisely performance sensitive...

i'd rather have more common codepaths than more conditioned, hence if there's no real need to only query this in case of --user instances i'd just always pick this up.

querying this value and using this value is two different things after all

Reply:
So the lines below introduced in 963b6b90 should also be common?